lib/syslog/writer.go

```diff
@@ -244,6 +285,26 @@ func (c bufferedConn) Close() error                { return c.conn.Close() }
 func (c bufferedConn) Flush() error                { return c.buf.Flush() }
 func (c bufferedConn) Write(b []byte) (int, error) { return c.buf.Write(b) }
 
+func writerCacheKey(network string, address string, socksProxy string) string {
+	// We don't currently include the TLS configuration in the cache key. For
+	// now, this is probably OK as the TLS configuration is unlikely to change.
+	return network + address + socksProxy
```

Comment:
Should we put separators between those values? Just to never risk running into conflicting names if one the combinations of different values happen to produce the same result?